hfer/server/app_logic.py

```diff
@@ -25,65 +27,73 @@ def __init__(
-        if not path.exists(save_dir):
-            makedirs(save_dir)
-        image_stem = Path(image_file).stem
-
-        img = Image.open(img_path)
-        face_image_files = []
-
-        for idx, face_coords in enumerate(result):
-            top, right, bottom, left = face_coords
-            crop_pic = img.crop((left, top, right, bottom))
-            image_file = image_stem + "_" + str(idx) + ".jpg"
-            save_path = Path(save_dir, image_file)
-            crop_pic.save(save_path)
-
-            face_image_files.append(image_file)
-
-        return face_image_files
-
-    def get_image(self, face_image_name, _type=None):
-        ## Consider using this and passing this around instead of the image path
-        img_path = Path(self.image_input_dir, face_image_name)
-        img = Image.open(img_path)
-        if _type == "json":
-            # Create a dictionary to store image information
-            image_info = {
-                "format": img.format,
-                "mode": img.mode,
-                "size": img.size,
-                "data": img.tobytes().decode(
-                    "latin1"
-                ),  # Convert bytes to string
-            }
-
-            # Convert dictionary to JSON
-            json_str = json.dumps(image_info)
-            return json_str
-
-        return img
-
-    def draw_faces_on_image(self, image_file, face_locations):
-        self.image_viewer.display_faces(image_file, face_locations)
+    def get_faces_from_image(self, image: np.array):
+
+        face_coords = self.extractor.extract_faces(image)
+        # Sort the faces by top value of bounding box.
+        # The height is used to make bands to put the boxes in.
+        height = image.shape[0]
+        face_coords = sorted(face_coords, key=lambda x: (x[0] // (height / 10), x[3]))
+        face_ids = []
+
+        for face_coord in face_coords:
+            top, right, bottom, left = face_coord
+            crop_pic = image[top:bottom, left:right]
+
+            face_id = uuid.uuid4().hex
+            face_ids.append(face_id)
+
+            self.faces_dict[face_id] = (crop_pic, face_coord)
+
+        return face_ids
+
+    # def get_image(self, face_image_name, _type=None):
+    #     ## Consider using this and passing this around instead of the image path
+    #     img_path = Path(self.image_input_dir, face_image_name)
+    #     img = Image.open(img_path)
+    #     if _type == "json":
+    #         # Create a dictionary to store image information
+    #         image_info = {
+    #             "format": img.format,
+    #             "mode": img.mode,
+    #             "size": img.size,
+    #             "data": img.tobytes().decode("latin1"),  # Convert bytes to string
+    #         }
+
+    #         # Convert dictionary to JSON
+    #         json_str = json.dumps(image_info)
+    #         return json_str
+
+    #     return img
+
+    def get_annotated_image(self, image, face_ids):
```

Comment:
This doesn't seem to need the face images... Wonder if we can easily refactor to only pass to it what it needs (coords) for efficiency. Not high-priority.